tools/scripts/refresh-ubuntu-amis.py

```diff
@@ -65,30 +64,69 @@ def update_tag_for_image_type(subconf: Dict[str, str], new_tag: str) -> bool:
     return True
 
 
-if __name__ == "__main__":
-    if len(sys.argv) < 2:
-        print(__doc__, file=sys.stderr)
-        sys.exit(1)
-
-    path = sys.argv[1]
-
+def update_bumpenvs_yaml(table: List[List[str]], path: str) -> None:
     with open(path) as f:
-        conf = yaml.safe_load(f)
+        bumpenvs_conf = yaml.safe_load(f)
 
-    for image_type, subconf in conf.items():
+    # All master-amis and bastion-amis in bumpenvs.yaml are updated.
+    # The agent-amis are updated after each rebuild of the environments repo.
+    for image_type, subconf in bumpenvs_conf.items():
         if image_type.endswith("_master_ami"):
             region = image_type[: -len("_master_ami")].replace("_", "-")
             # Master AMIs are based on Focal.
-            new_ami = get_ubuntu_ami("focal", region)
+            new_ami = get_ubuntu_ami(table, "focal", region)
             if new_ami is not None:
                 update_tag_for_image_type(subconf, new_ami)
 
         if image_type.endswith("_bastion_ami"):
             region = image_type[: -len("_bastion_ami")].replace("_", "-")
             # Bastion AMIs are based on Focal.
-            new_ami = get_ubuntu_ami("focal", region)
+            new_ami = get_ubuntu_ami(table, "focal", region)
             if new_ami is not None:
                 update_tag_for_image_type(subconf, new_ami)
 
     with open(path, "w") as f:
-        yaml.dump(conf, f, sort_keys=True)
+        yaml.dump(bumpenvs_conf, f, sort_keys=True)
+
+
+def update_packer_json(table: List[List[str]], path: str) -> None:
+    with open(path) as f:
+        packer_conf = json.load(f)
+
+    # There are two specific keys we set in the environments-packer.json file.
+    new_ami = get_ubuntu_ami(table, "focal", "us-west-2")
+    packer_conf["variables"]["aws_base_image"] = new_ami
+    new_ami = get_ubuntu_ami(table, "focal", "us-gov-west-1")
+    packer_conf["variables"]["gov_aws_base_image"] = new_ami
+
+    with open(path, "w") as f:
+        json.dump(packer_conf, f, indent="  ")
+        # json.dump() leaves off the final newline.
+        f.write("\n")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawDescriptionHelpFormatter
+    )
+    parser.add_argument("--bumpenvs-yaml", metavar="/PATH/TO/BUMPENVS.YAML")
+    parser.add_argument("--packer-json", metavar="/PATH/TO/ENIRONMENTS-PACKER.JSON")
+    args = parser.parse_args()
+
+    if not args.bumpenvs_yaml and not args.packer_json:
+        parser.print_help(sys.stderr)
+        sys.exit(1)
+
+    # Do one fetch for all the ubuntu amis.
+    # Note that this is not a public endpoint, but it's the only one that has the gov amis.
```

Comment:
```suggestion
    # Note that this is not intended as a machine-readable endpoint, but it's the only one that has the gov amis.
```

Reply:
I mean... it is machine readable (it's ready by a javascript page) but that's different than a public api, which you could expect to be more stable over time.